pkg/gofr/http/middleware/apikey_auth_test.go

```diff
@@ -24,26 +34,34 @@ func Test_ApiKeyAuthMiddleware(t *testing.T) {
 	}
 
 	testCases := []struct {
-		desc         string
-		validator    func(apiKey string) bool
-		apiKey       string
-		responseCode int
-		responseBody string
+		desc                string
+		validatorFunc       func(akiKey string) bool
+		validatorFuncWithDB func(c *container.Container, apiKey string) bool
+		apiKey              string
+		responseCode        int
+		responseBody        string
 	}{
-		{"missing api-key", nil, "", 401, "Unauthorized: Authorization header missing\n"},
-		{"invalid api-key", nil, "invalid-key", 401, "Unauthorized: Invalid Authorization header\n"},
-		{"valid api-key", nil, "valid-key-1", 200, "Success"},
-		{"another valid api-key", nil, "valid-key-2", 200, "Success"},
-		{"custom validator valid key", validator, "valid-key", 200, "Success"},
-		{"custom validator in-valid key", validator, "invalid-key", 401, "Unauthorized: Invalid Authorization header\n"},
+		{"missing api-key", nil, nil, "", 401, "Unauthorized: Authorization header missing\n"},
+		{"invalid api-key", nil, nil, "invalid-key", 401, "Unauthorized: Invalid Authorization header\n"},
+		{"valid api-key", nil, nil, validKey1, 200, "Success"},
+		{"another valid api-key", nil, nil, validKey2, 200, "Success"},
+		{"custom validatorFunc valid key", validator, nil, validKey1, 200, "Success"},
+		{"custom validatorFuncWithDB valid key", nil, validatorWithDB, validKey2, 200, "Success"},
+		{"custom validatorFuncWithDB in-valid key", nil, validatorWithDB, "invalid-key", 401, "Unauthorized: Invalid Authorization header\n"},
```

Comment:
Test case can be added where both of these are being registered?
What would be used in that case?

i.e user is calling both of these methods:
```
	EnableAPIKeyAuth()
	EnableAPIKeyAuthWitValidator()
```

Reply:
Added